### PR TITLE
Update to shakapacker gem

### PR DESCRIPTION
**Pull request by justin808**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/27)

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
+      ]
     ].filter(Boolean),
   }
 
-  resultConfig.plugins = [...resultConfig.plugins, ...changesOnDefault.plugins ];
+  resultConfig.presets = [...resultConfig.presets, ...changesOnDefault.presets]
+  resultConfig.plugins = [...resultConfig.plugins, ...changesOnDefault.plugins ]
 
-  return resultConfig;
+  return resultConfig
```
Comment (justin808) [inline babel.config.js:29]: These changes are needed if you're using React JSX.

File: config/webpack/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  const path = resolve(__dirname, `${env.nodeEnv}.js`)
+  if (existsSync(path)) {
+    console.log(`Loading ENV specific webpack configuration file ${path}`)
+    return require(path)
+  } else {
+    return webpackConfig
+  }
+}
+
+module.exports = envSpecificConfig()
```
Comment (justin808) [inline config/webpack/webpack.config.js:15]: We now have a single configuration file.

Here's the shim for loading the older style of a file per value of NODE_ENV, the:

* `/config/webpack/test.js`
* `/config/webpack/development.js`
* `/config/webpack/production.js`

File: package.json [hunk 3]
```diff
@@ -1,26 +1,39 @@
...
+    "webpack-cli": "^4.9.1",
+    "webpack-merge": "^5.8.0",
+    "webpack-sources": "^3.2.3"
   },
   "devDependencies": {
     "@pmmmwh/react-refresh-webpack-plugin": "^0.5.1",
     "fork-ts-checker-webpack-plugin": "^6.0.1",
     "react-refresh": "^0.9.0",
-    "webpack-dev-server": "^4.2.1"
+    "webpack-dev-server": "^4.7.3"
```
Comment (justin808) [inline package.json:37]: Run the command:

```
  yarn add @babel/core @babel/plugin-transform-runtime @babel/preset-env @babel/runtime babel-loader compression-webpack-plugin pnp-webpack-plugin terser-webpack-plugin webpack webpack-assets-manifest webpack-cli webpack-merge webpack-sources webpack-dev-server
```

Comment (Judahmeek) [reply package.json:37]: @justin808 run the command for what?

**Pull request merged**